src/Cassandra/Observers/RequestTracker/RequestTrackerObserver.cs

```diff
@@ -0,0 +1,78 @@
+﻿//
+//      Copyright (C) DataStax Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+using System;
+using System.Threading.Tasks;
+using Cassandra.Observers.Abstractions;
+using Cassandra.Requests;
+
+namespace Cassandra.Observers.RequestTracker
+{
+    internal class RequestTrackerObserver : IRequestObserver
+    {
+        private readonly IRequestTracker _requestTracker;
+
+        public RequestTrackerObserver(IRequestTracker requestTracker)
+        {
+            _requestTracker = requestTracker;
+        }
+
+        public async Task OnNodeRequestError(
+            Host host,
+            RequestErrorType errorType,
+            RetryDecision.RetryDecisionType decision,
+            RequestTrackingInfo r,
+            Exception ex)
+        {
+            var hostInfo = new HostTrackingInfo { Host = host };
+            
+            await _requestTracker.OnNodeErrorAsync(r, hostInfo, ex).ConfigureAwait(false);
```

Comment:
There's a minor optimization that can be done here, I know this was true some years ago but I don't know if modern .NET versions have made this irrelevant:

In cases like this where there's only one `await` and it's the last "statement" of the method then we can just remove `async await` and just return the task itself.

```
public Task OnNodeRequestError(
            Host host,
            RequestErrorType errorType,
            RetryDecision.RetryDecisionType decision,
            RequestTrackingInfo r,
            Exception ex)
        {
            var hostInfo = new HostTrackingInfo { Host = host };

            return _requestTracker.OnNodeErrorAsync(r, hostInfo, ex);
```

Reply:
This can also be done for the other methods of this class. Again it's a very minor thing and not that important but it does make it so there's one less `async/await` state machine that the compiler generates (unless modern .NET versions make this optimization automatically).

Reply:
I honestly don't know if the latest .NET versions have made the optimization but given that the package is .NET Standard, which can be executed by a .NET Framework runtime, I think it's safer to make the optimization.
Changed it in the last commit